services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/presignedurl/PresignedUrlGetObjectRequestMarshallerTest.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.internal.presignedurl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import java.net.URL;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.protocols.xml.AwsXmlProtocolFactory;
+import software.amazon.awssdk.services.s3.internal.presignedurl.model.PresignedUrlGetObjectRequestWrapper;
+
+class PresignedUrlGetObjectRequestMarshallerTest {
+
+    private PresignedUrlGetObjectRequestMarshaller marshaller;
+    private AwsXmlProtocolFactory mockProtocolFactory;
+    private URL testUrl;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        mockProtocolFactory = mock(AwsXmlProtocolFactory.class);
+        marshaller = new PresignedUrlGetObjectRequestMarshaller(mockProtocolFactory) {
+            @Override
+            public SdkHttpFullRequest marshall(PresignedUrlGetObjectRequestWrapper request) {
```

Comment:
Because we override the marshal method here, are we actually testing the `PresignedUrlGetObjectRequestMarshaller` class? Isn't the test code not gonna call the real method, but instead this one?